tmuxPlugins.extrakto: unstable-2021-04-04 -> 0-unstable-2024-08-25

uimataso opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

- Add dependency `wl-clipboard` since extrakto added support for Wayland via `wl-copy`
- Injected Python3 executable to Python scripts shebang, since the way extrakto called Python scripts has changed

This is my first pr, tho I have read the contributing guide, but I am still not too comfortable doing this :/

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
* this takes forever for my wifi to run :(
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Commit:
tmuxPlugins.extrakto: unstable-2021-04-04 -> 0-unstable-2024-08-26

The Tmux plugin extrakto has added clipboard support for Wayland via `wl-copy`, so I added the `wl-clipboard` package as a dependency. The way extrakto executes Python scripts has also changed, so I directly changed the shebang of the Python script to the Python3 path to make sure the Python path is correctly set.

fnune commented:
Looks like I did the same thing again over here https://github.com/NixOS/nixpkgs/pull/343088

fnune commented on pkgs/misc/tmux-plugins/default.nix:
```diff
@@ -216,22 +216,23 @@ in rec {
   extrakto = mkTmuxPlugin {
     pluginName = "extrakto";
-    version = "unstable-2021-04-04";
+    version = "0-unstable-2024-08-26";
```
Nit: the commit is from Aug 25, so the version should be `unstable-2024-08-25` https://github.com/laktak/extrakto/commit/bf9e666f2a6a8172ebe99fff61b574ba740cffc2

Just curious: why `0-`?

uimataso replied:
I read [this doc](https://github.com/NixOS/nixpkgs/tree/master/pkgs#versioning), and it say versioning must start with digit. I just follows the example under the third point.

On my page, GitHub say `Commits on Aug 26, 2024`, so maybe it is timezone  issue?

uimataso replied:
Idk how to read the right time tho

fnune commented:
@kidd?

fnune commented:
How can we get this merged? It seems like the maintainer is no longer active. There is this PR and this other one: https://github.com/NixOS/nixpkgs/pull/343088 which it would be great to merge.

fnune commented:
https://github.com/NixOS/nixpkgs/pull/347687

Trying to get access here

fpletz commented on pkgs/misc/tmux-plugins/default.nix:
```diff
@@ -216,22 +216,23 @@ in rec {
-        [ pkgs.fzf pkgs.python3 pkgs.xclip ]
-        )}
-    done
+      for f in extrakto.py extrakto_plugin.py; do
+        sed -i -e 's|#!/usr/bin/env python3|#!${pkgs.python3}/bin/python3|g' $target/$f
+      done
```
Using `patchShebangs` is easier and prone to breakge than the `sed` invocation.
```suggestion
     patchShebangs extrakto.py extrakto_plugin.py
```

fpletz commented on pkgs/misc/tmux-plugins/default.nix:
```diff
@@ -216,22 +216,23 @@ in rec {
+      rev = "bf9e666f2a6a8172ebe99fff61b574ba740cffc2";
+      sha256 = "sha256-kIhJKgo1BDTeFyAPa//f/TrhPfV9Rfk9y4qMhIpCydk=";
     };
     nativeBuildInputs = [ pkgs.makeWrapper ];
```
Needed for `patchShebangs` to find the python interpreter.
```suggestion
    nativeBuildInputs = [ pkgs.makeWrapper ];
    buildInputs = [ pkgs.python3 ];
```

uimataso committed:
Apply suggestions from code review that use `patchShebangs` instead of `sed`

Co-authored-by: Franz Pletz <fpletz@fnordicwalking.de>

uimataso committed:
fix: fix the updated version from 2024-08-25 to 2024-08-26

Due to the timezone, I misread the upstream commit as 2024-08-26, but it actually is 2024-08-25

uimataso commented:
Hi, I apply the review and fix the version that @fnune points out is wrong.

BTW, I am not familiar with GitHub review and NixOS contribution, so if anything I should do or read, just tell me :)

fpletz reviewed (approved):
Thanks, code looks fine now. Just one more thing for the future: Please squash your commits so that it's just a single commit. I can squash them via Github so that's no issue for now. But if you did more then just a version bump it would be more than one final commit.

> BTW, I am not familiar with GitHub review and NixOS contribution, so if anything you think I should do or read, just tell me :)

Reading https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md should give you the basics and the other stuff like `patchShebangs` are things you will get to know when you are touch more code. So no worries there. :smiley:

fnune reviewed (approved):
Ran a quick local test on Wayland and it's working fine. THX :+1:
